pywinauto/hooks.py

```diff
@@ -0,0 +1,276 @@
+                 91: 'Lwin',
+                 92: 'Rwin',
+                 93: 'App',
+                 95: 'Sleep',
+                 96: 'Numpad0',
+                 97: 'Numpad1',
+                 98: 'Numpad2',
+                 99: 'Numpad3',
+                 100: 'Numpad4',
+                 101: 'Numpad5',
+                 102: 'Numpad6',
+                 103: 'Numpad7',
+                 104: 'Numpad8',
+                 105: 'Numpad9',
+                 106: 'Multiply',
+                 107: 'Add',
+                 109: 'Subtract',
+                 110: 'Decimal',
+                 111: 'Divide',
+                 112: 'F1',
+                 113: 'F2',
+                 114: 'F3',
+                 115: 'F4',
+                 116: 'F5',
+                 117: 'F6',
+                 118: 'F7',
+                 119: 'F8',
+                 120: 'F9',
+                 121: 'F10',
+                 122: 'F11',
+                 123: 'F12',
+                 144: 'Numlock',
+                 160: 'Lshift',
+                 161: 'Rshift',
+                 162: 'Lcontrol',
+                 163: 'Rcontrol',
+                 164: 'Lmenu',
+                 165: 'Rmenu',
+                 186: 'Oem_1',
+                 187: 'Oem_Plus',
+                 188: 'Oem_Comma',
+                 189: 'Oem_Minus',
+                 190: 'Oem_Period',
+                 191: 'Oem_2',
+                 192: 'Oem_3',
+                 219: 'Oem_4',
+                 220: 'Oem_5',
+                 221: 'Oem_6',
+                 222: 'Oem_7',
+                 1001: 'mouse left',  # mouse hotkeys
+                 1002: 'mouse right',
+                 1003: 'mouse middle',
+                 1000: 'mouse move',  # single event hotkeys
+                 1004: 'mouse wheel up',
+                 1005: 'mouse wheel down',
+                 1010: 'Ctrl',  # merged hotkeys
+                 1011: 'Alt',
+                 1012: 'Shift',
+                 1013: 'Win'}
+
+    event_types = {0x100: 'key down',  # WM_KeyDown for normal keys
+                   0x101: 'key up',  # WM_KeyUp for normal keys
+                   0x104: 'key down',  # WM_SYSKEYDOWN, used for Alt key.
+                   0x105: 'key up',  # WM_SYSKEYUP, used for Alt key.
+                   }
+
+    WH_KEYBOARD_LL = 0x00D
+    WH_MOUSE_LL = 0x0E
+
+    def __init__(self):
+        self.handler = 0
+        self.pressed_keys = []
+        self.keyboard_id = None
+        self.mouse_id = None
+
+    def hook(self, keyboard=True, mouse=False):
+        """Hook mouse and/or keyboard events"""
+        if not mouse and not keyboard:
+            return
+
+        if keyboard:
+            def keyboard_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when keyboard low level event was catched"""
+                key_code = kb_data_ptr[0]
+                current_key = self.ID_TO_KEY[key_code]
+                event_type = self.event_types[event_code]
```

Comment:
Generally, any input from the outer world can't be trusted so any of these dictionaries translations may fail. Not sure if we have a time for an extensive validation here so I'd suggest to guard with ```try``` and use ```finally``` to make sure we pass the event to the next-in-chain hook even if we failed to handle it.
The same treatment needed for ```mouse_low_level_handler```